nightingale_dispatcher/src/nightingale_dispatcher/navigate_task.py

```diff
@@ -1,18 +1,30 @@
 #!/usr/bin/env python3
 
 import rospy
-
 from nightingale_dispatcher.task import Task
+import actionlib
+from actionlib_msgs.msg import GoalStatus
+from nightingale_msgs.msg import RoomRunnerAction, RoomRunnerActionGoal
 
 
 class NavigateTask(Task):
-    # Go to position with given context
-    # TODO define parameters
-    def __init__(self, *args, **kwargs):
-        raise NotImplementedError()
+    def __init__(self):
+        action_name = "/nightingale_navigation/room_runner"
+        self.action_client = actionlib.SimpleActionClient(action_name, RoomRunnerAction)
+        rospy.loginfo(f"Waiting for {action_name} action server")
+        if self.action_client.wait_for_server():
+            rospy.loginfo(
+                f"Found the {action_name} action server",
+            )
+        else:
+            rospy.logfatal(
+                f"Failed to find the {action_name} action server",
+            )
 
-    # TODO define inputs
-    def execute(self, *args, **kwargs):
-        # Check if there are intermediate steps to travel to the destination
-        # Call room_runner to drive to pose
-        raise NotImplementedError()
+    def execute(self, room_number, sublocation):
+        goal = RoomRunnerActionGoal()
```

Comment:
Do we need to check if there are any doors in the path to get to the desired room?

Reply:
see the comment in line 50 of the other file. if we start handling doors, we must then ask then it becomes:
nav_to_door -> door handling node -> nav_to_bedside instead of just nav to bedside